File: zykh_station_app/frontend/src/api/network.js
import { apiGet, apiPost } from "./client.js";

export function loadNetworkStatus() {
  return apiGet("/api/network/status");
}

export function setNetworkMode(mode) {
  return apiPost("/api/network/mode", { mode });
}

export function startQsm4g() {
  return apiPost("/api/network/start-4g", {});
}

File: zykh_station_app/backend/app/routers/network.py
from __future__ import annotations

from fastapi import APIRouter
from pydantic import BaseModel

from ..services.network_service import NetworkService

router = APIRouter(prefix="/api/network", tags=["network"])


class NetworkModeRequest(BaseModel):
    mode: str


@router.get("/status")
def network_status() -> dict[str, object]:
    return NetworkService().status()


@router.post("/mode")
def set_network_mode(request: NetworkModeRequest) -> dict[str, object]:
    return NetworkService().set_mode(request.mode)


@router.post("/start-4g")
def start_4g_network() -> dict[str, object]:
    return NetworkService().start_4g()
